src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -231,7 +231,22 @@ private void Enlarge()
                 Debug.Assert(!_isLocked, "Appending to a locked database");
 
                 byte[] toReturn = _data;
-                _data = ArrayPool<byte>.Shared.Rent(toReturn.Length * 2);
+
+                // Allow the data to grow up to maximum possible capacity (~2G bytes) before encountering overflow.
```

Comment:
Shouldn't this also verify that `toReturn.Length` is less than the max length?

Reply:
Should it throw an `OutOfMemoryException` if they're equal? With message like "Data capacity exceeded supported range"

Reply:
That's about right. But other than throwing an exception explicitly you could try setting `newCapacity` to `int.MaxValue` or something so that the buffer pool throws the right exception for you.

Reply:
Done!